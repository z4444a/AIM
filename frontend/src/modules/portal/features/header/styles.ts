import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';

export type Style = WithStyles<'appBar' | 'toolbar' | 'menuButton' | 'emptySpace'>;

export const styles = (theme: Theme) =>
  createStyles({
    appBar: {
      position: 'fixed',
      zIndex: 1400,
    },
    toolbar: theme.mixins.toolbar,
    menuButton: {
      marginRight: 36,
    },
    emptySpace: {
      flexGrow: 1,
    },
  });
