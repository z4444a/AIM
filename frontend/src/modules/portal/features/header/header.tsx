import React, { ReactNode } from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import { mapDispatchToProps, mapStateToProps } from './index';
import { Style } from './styles';

export type Props = ReturnType<typeof mapStateToProps> &
  ReturnType<typeof mapDispatchToProps> &
  Style;

export class Header extends React.PureComponent<Props> {
  public render(): ReactNode {
    const { classes, name, logout, toggleSideNav } = this.props;

    return (
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="Open drawer"
            onClick={toggleSideNav}
            className={classes.menuButton}
          >
            <MenuIcon />
          </IconButton>
          {this.props.children}
          <div className={classes.emptySpace} />
          <h4>{name}</h4>
          <IconButton onClick={logout}>
            <ExitToAppIcon />
          </IconButton>
        </Toolbar>
      </AppBar>
    );
  }
}
