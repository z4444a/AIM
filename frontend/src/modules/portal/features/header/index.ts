import { connect } from 'react-redux';
import withStyles from '@material-ui/core/styles/withStyles';
import { styles } from './styles';
import actions from '../../../../redux/actions';
import { Header } from './header';
import { RootState } from 'typesafe-actions';

export const mapStateToProps = (state: RootState) => ({
  name: state.tokens && state.tokens.user ? state.tokens.user.fullName : '',
});

export const mapDispatchToProps = (dispatch: Function) => ({
  logout: () => dispatch(actions.logout.logout()),
  toggleSideNav: () => dispatch(actions.portal.togglePortalSideNavPanel()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(Header));
