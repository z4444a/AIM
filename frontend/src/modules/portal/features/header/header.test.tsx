import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { Header, Props } from './header';
import sinon from 'sinon';

describe('AppHeader', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const props: Props = {
      classes: {
        appBar: 'appBar',
        toolbar: 'toolbar',
        menuButton: 'menuButton',
        emptySpace: 'emptySpace',
      },
      name: 'userName',
      toggleSideNav: sinon.spy(),
      logout: sinon.spy(),
    };
    wrapper = shallow(<Header {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
