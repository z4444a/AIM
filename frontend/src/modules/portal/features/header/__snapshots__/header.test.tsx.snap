// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render correctly 1`] = `
<WithStyles(AppBar)
  className="appBar"
  position="fixed"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Open drawer"
      className="menuButton"
      color="inherit"
      onClick={[Function]}
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <div
      className="emptySpace"
    />
    <h4>
      userName
    </h4>
    <WithStyles(IconButton)
      onClick={[Function]}
    >
      <pure(ExitToAppIcon) />
    </WithStyles(IconButton)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
